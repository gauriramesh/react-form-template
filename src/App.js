import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';


class App extends Component {
    constructor(props) {
        super(props);

        this.state = {
            name: '',
            email: '',
            phone: '',
            nameError: '',
            emailError: '',
            phoneError: ''
        }

        this.validateEmail = this.validateEmail.bind(this);
        this.validateName = this.validateName.bind(this);
        this.validatePhone = this.validatePhone.bind(this);
        this.updateState = this.updateState.bind(this);
    };



    validateEmail(input) {
        //language=JSRegexp
        //TODO: Fix this regex, doesn't actually take everything into account.
        var regexp = /^([a-zA-Z0-9]+)@(.+)(\.)(com|org|net|edu)/;
        return regexp.test(input);
    }

    validateName(input) {
        if(input.length >= 2) {
            return true;
        } return false;
    }

    validatePhone(input) {
        var regexp = /[0-9]{10}/;
        return regexp.test(input);
    }



    //TODO: Make sure class name and state name are the same.
    //updates the state by taking in an event, a validator function (i.e validateName, validatePhone), and an error name to update the state of.
    updateState(e, validatorFunction, errorName) {
            if (validatorFunction === true) {
                this.setState({[e.target.classList[0]]: e.target.value});
                this.setState({[errorName]: 'Valid'});
                document.getElementById(e.target.classList[0]).style.color = "green";
            }
            else if (e.target.value.length === 0) {
                this.setState({[errorName]: 'Empty submission'});
                document.getElementById(e.target.classList[0]).style.color = "red";
            }
            else {
                this.setState({[errorName]: 'Invalid form input'});
                document.getElementById(e.target.classList[0]).style.color = "red";
                //Test
            }
    }

    //create custom error method.

    //Maybe create a validator component that changes depending on the input?


  render() {
    // let phoneClass = '';
    // if (this.state.phoneError !== 'Valid') {
    //     phoneClass = 'successful-phone';
    // } else {
    //     phoneClass = 'failing-phone';
    // }
    return (
      <div className="App">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h2>React Form</h2>
        </div>
          <form>
              Name:
              <input className="name" type="text" id="input-text-sm"
                     onChange={(event) => this.updateState(event, this.validateName(event.target.value), "nameError")}/>
              <p id="name">  {this.state.nameError} </p> <br/>
              Email:
              <input className="email" type="text"
                     onChange={(event) => this.updateState(event, this.validateEmail(event.target.value), "emailError")}/>
              <p id="email">  {this.state.emailError} </p> <br/>
              Phone:
              <input className="phone" /*{phoneClass}*/ type="text"
                     onChange={(event) => this.updateState(event, this.validatePhone(event.target.value), "phoneError")}/>
              <p id="phone">  {this.state.phoneError} </p> <br/>
              <input type="submit"/>
          </form>
      </div>
    );
  }
}

export default App;
