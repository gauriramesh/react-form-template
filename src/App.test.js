import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

test('two plus two is four', () => {
  expect(2 + 2).toBe(4);
});

test('object assignment', () => {
    const data = {one: 1};
    data['two'] = 2;
    expect(data).toEqual({one: 1, two: 2});
});

test('there is no I in team', () => {
   expect('team').not.toMatch(/I/);
   //TODO: case sensitivity?
});

test('but there is a "stop" in Christoph', () => {
   expect('Christoph').toMatch(/stop/);
});

//Testing Arrays
const friends = [
    "Gauri",
    "Troy",
    "Allison",
    "Reid",
    "Joe",
];

test('the friends has Allison in it', () => {
    expect(friends).toContain('Allison');
});



//Tests I could write for my form:
// TODO: Testing all of the regex: if put in an invalid input, check that it updates the state as such.
// TODO: Reloads on submit

it('snapshots the page correctly', () => {
    const component = renderer.create(<App/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});






