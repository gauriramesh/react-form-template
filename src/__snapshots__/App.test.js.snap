// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots the page correctly 1`] = `
<div
  className="App"
>
  <div
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h2>
      React Form
    </h2>
  </div>
  <form>
    Name:
    <input
      className="name"
      onChange={[Function]}
      type="text"
    />
    <p
      id="name"
    >
        
      
       
    </p>
     
    <br />
    Email:
    <input
      className="email"
      onChange={[Function]}
      type="text"
    />
    <p
      id="email"
    >
        
      
       
    </p>
     
    <br />
    Phone:
    <input
      className="phone"
      onChange={[Function]}
      type="text"
    />
    <p
      id="phone"
    >
        
      
       
    </p>
     
    <br />
    <input
      type="submit"
    />
  </form>
</div>
`;
